### DISCUSSION TITLE
Am I impersonating?

### DISCUSSION THREAD

**Discussion by BarbecueSilver** [accounts-js/accounts#1195] [Q&A] [answered]
Hej there,

thanks at first for this awesome project. :+1: 

---

**Question**
When I had run `impersonate({...})`, how do I know that I am actually impersonating another user?

**Context**
I want to setup an impersonation feature in the way that you are visually aware of impersonating a different user. Also I want to show up a button which triggers `stopImpersonate()`, so that the user must not logout fully and login again.
Hope I didn't miss any information in the docs or code.

**Comment by BarbecueSilver** [comment, marked as answer]
I maybe made some progress here. 

Did the following on the server side:
```
  const currentSession = await accountsServer.findSessionByAccessToken(authToken);
```

Via TypeScript I get the return type of a `Session`:
```
export interface Session {
  id: string;
  userId: string;
  token: string;
  valid: boolean;
  userAgent?: string | null;
  ip?: string | null;
  createdAt: string;
  updatedAt: string;
}
```

When checking the `currentSession`, I have some additional data.
`extraData` is missing either in the interface or in the return type of `findSessionByAccessToken`.
```
{
  ...
  // Additional to the mentioned properties above:
  extraData: { impersonatorUserId: 'someId123321' },
}
```

---

Shall I create an issue for it?

**Reply by pradel** [reply]
Yes please, thanks for digging into it :)

**Reply by BarbecueSilver** [reply]
@pradel 
Created an [issue](https://github.com/accounts-js/accounts/issues/1197) and a [fix](https://github.com/accounts-js/accounts/pull/1196).

**Reply by BarbecueSilver** [reply]
To mention it here as well: Opened a PR to add some [more information to the docs](https://github.com/accounts-js/accounts/pull/1199).
